libr/anal/p/anal_arm_hacks.inc

```diff
@@ -1,5 +1,126 @@
+static int hack_handle_dp_reg(ut32 insn, RAnalOp *op) {
+	const ut8 op0 = (insn >> 30) & 0x1;
+	const ut8 op1 = (insn >> 28) & 0x1;
+	const ut8 op2 = (insn >> 21) & 0xf;
+
+	// Data-processing (2 source)
+	if (op0 == 0 && op1 == 1 && op2 == 0x6) {
+		const ut8 sf = (insn >> 31) & 0x1;
+		const ut8 S = (insn >> 29) & 0x1;
+		const ut8 opcode = (insn >> 10) & 0x1f;
+		if (sf == 1 && S == 0 && opcode == 4) {
```

Comment:
all the conditions are checkinf if sf == 1, can you just group them all in a nested way instead of checking the same on every line?. there's also a subgropup for opcode == 0, so the amount of conditions executed can be reduced a bit

Reply:
agree

Reply:
> all the conditions are checkinf if sf == 1, can you just group them all in a nested way instead of checking the same on every line?. 

The idea is that every level of nesting represents a level of encoding in the ARM reference manual so it's easier to follow if you want to look at it side-by-side
> there's also a subgropup for opcode == 0, so the amount of conditions executed can be reduced a bit

I trust the compiler's ssa to actually calculate these comparisons once, but whatever ¯\_(ツ)_/¯